### ISSUE TITLE
(feat): authenticate.idp.clientSecret from preexisting secret

### ISSUE THREAD

**Issue by lukasmrtvy** [pomerium/pomerium-helm#104]
Would be nice to be able to set `authenticate.idp.clientSecret` from preexisting secret.
Thanks

**Comment by travisgroth** [comment]
Hi - what's your use case for this?

**Comment by lukasmrtvy** [comment]
* bootstraping secrets ( prior to deployments ) 
* secret is created with idp itself. 
* ever major helm chart can reference sensitive values from pre existing secret
* gitops pattern ( helmfile, kustomize )

**Comment by lukasmrtvy** [comment]
will You accept PR for adding `authenticate.idp.clientSecretRef` ?

**Comment by travisgroth** [comment]
I haven't seen this pattern enabled very frequently.  Usually a user takes a helm chart's values or supplies an entire secret.  I'm curious why you want to source just the clientSecret from outside the chart.

That said, I could see doing this in a generic way by augmenting `extraEnv` to accept either `value` or `valueFrom` fields.  Then we could lean on https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#envvarsource-v1-core to pull data in from wherever the operator wants.

Does that make sense / work for your use case?  I think this can be done in a way that is backwards compatible with no extra top level values.  Would be happy to see a PR for that.

**Comment by lukasmrtvy** [comment]
Its common practice between all major mature charts, where You need to prebootstrap secret somehow.

Of course that I can work similar to stable/grafana chart, for example: `--set envValueFrom.GF_DATABASE_PASSWORD.secretKeyRef.name=custom-grafana-secret `
, but it will not work for https://github.com/pomerium/pomerium-helm/blob/master/charts/pomerium/templates/secret.yaml#L68
AFAIK There is no way for checking value name in helm, like regex filtering and its really pain to render config file like that.

I am suggesting to use environment variables for configuring pomerium in helm charts and secret mounting for policies.

**Comment by travisgroth** [comment]
> Its common practice between all major mature charts, where You need to prebootstrap secret somehow.

There's no requirement to prebootstrap this value.  You can set it at chart install time through whatever your automated process looks like.  

Can you provide an example of a chart that provides what you're looking for so I can get a sense of it?  Perhaps I'm not understanding what you're trying to do.  

>Of course that I can work similar to stable/grafana chart, for example: --set envValueFrom.GF_DATABASE_PASSWORD.secretKeyRef.name=custom-grafana-secret
, but it will not work for https://github.com/pomerium/pomerium-helm/blob/master/charts/pomerium/templates/secret.yaml#L68

I'm sorry but I'm not clear on the problem here.  This was the interface for my proposal:

Given:

secret:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: MY_EXISTING_SECRET
type: Opaque
data:
  MY_EXISTING_IDP_CLIENT_SECRET: [BASE64 ENCODED CLIENT SECRET]
```

pomerium values:
```yaml
extraEnv:
  - name: IDP_CLIENT_SECRET
    valueFrom:
      secretKeyRef:
        name: MY_EXISTING_SECRET
        key: MY_EXISTING_IDP_CLIENT_SECRET
```

That should source the value of `MY_EXISTING_IDP_CLIENT_SECRET` from secret `MY_EXISTING_SECRET` as `IDP_CLIENT_SECRET` env var for pomerium, and this would work for any valid `EnvVarSource` selector.

You can configure the chart with env vars almost entirely if you feel like using `extraEnv` and ignoring the rest of the `config` block.  The ability to use `valueFrom` would enable this more substantially.  

The chart also supports sourcing the config file from an existing secret which you can generate in whatever way makes sense for your environment.

**Comment by lukasmrtvy** [comment]
Good. Lets implement it. Only thing, again, https://github.com/pomerium/pomerium-helm/blob/master/charts/pomerium/templates/secret.yaml#L68 should be handled properly ..
`extraEnv.IDP_CLIENT_SECRET.valueFrom.secretKeyRef.name` or `authenticate.idp.clientSecret` should be considered as required.

**Comment by travisgroth** [comment]
This should be closed by #125
